src/peer_connection/mod.rs

```diff
@@ -447,27 +447,48 @@ impl RTCPeerConnection {
                 // if t.stopping && !t.stopped {
                 // 	return true
                 // }
-                let m = get_by_mid(t.mid().await.as_str(), local_desc);
+                let mid = t.mid().await;
+                let m = get_by_mid(&mid, local_desc);
                 // Step 5.2
                 if !t.stopped.load(Ordering::SeqCst) && m.is_none() {
                     return true;
                 }
                 if !t.stopped.load(Ordering::SeqCst) {
                     if let Some(m) = m {
                         // Step 5.3.1
```

Comment:
See https://www.w3.org/TR/webrtc/#ref-for-dfn-check-if-negotiation-is-needed-1 for the steps outlined by the specification